pennylane/plugins/expt_tensornet.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Experimental simulator plugin based on tensor network contractions
+"""
+import functools
+
+import numpy as np
+import tensornetwork as tn
+
+from pennylane.utils import _flatten
+from pennylane import Device
+from pennylane.plugins.default_qubit import I, X, Y, Z, H, CNOT, SWAP, CZ, S, T, CSWAP, \
+    Rphi, Rotx, Roty, Rotz, Rot3, CRot3, CRotx, CRoty, CRotz, CRot3, hermitian, identity, \
+    unitary
```

Comment:
Another reason to move the matrix representations into the actual PennyLane operation classes re: #297

Reply:
Agreed